### PR TITLE
WDDR PHY Prep API

### PR DESCRIPTION
**Pull request by wavious-wpatty**
Fixes:
  - None

Features:
  - WDDR Prep API updated to take MSR as parameter
    rather than implicitly using next MSR. This
    made it easier to pair a call to this function
    with a call to software frequency switch function.

### CODE REVIEW

File: dev/fsw/device.c [hunk 1]
```diff
@@ -85,6 +85,16 @@ void fsw_switch_to_dfi_mode(fsw_dev_t *dev)
     dev->mode = FSW_MODE_DFI;
 }
 
+wddr_msr_t fsw_get_current_msr(fsw_dev_t *dev)
```
Comment (wavious-jbasista) [inline dev/fsw/device.c]: Do we need to pass fsw_dev_t into this function?  It seems like it is always unused.

File: include/drivers/fsw/driver.h [hunk 2]
```diff
@@ -75,7 +75,17 @@ void fsw_ctrl_set_prep_done_reg_if(bool done);
...
-void fsw_ctrl_set_post_work_done(bool override, bool done);
+void fsw_ctrl_set_post_work_done_reg_if(bool override, bool done);
+
+/**
+ * @brief   HW Frequency Switch Get Current MSR Register Interface
+ *
+ * @details Gets the current Mode Switch Register (MSR) in the PHY.
+ *
+ * @return  returns current Mode Switch Register value.
+ * @retval  Possible MSRs values are 0 and 1.
```
Comment (wavious-jbasista) [inline include/drivers/fsw/driver.h:86]: I know the values are 0 and 1, but should we use WDDR_MSR_0 and WDDR_MSR_1 here to check against as the return vals?

File: dev/fsw/device.c (hunk 1 above)
Comment (wavious-wpatty) [reply dev/fsw/device.c]: I left it way to match signature at the device level. We typically always pass in the device so I kept that here even if it's unused. I will update to mark parameter as unused.

File: include/drivers/fsw/driver.h (hunk 2 above)
Comment (wavious-wpatty) [reply include/drivers/fsw/driver.h:86]: We can but I wanted to keep the driver layer a little less abstracted and have the device level work with wddr_msr_t enumeration.

File: dev/fsw/device.c (hunk 1 above)
Comment (wavious-jbasista) [reply dev/fsw/device.c]: I am ok with leaving as is then, makes sense to pass the device into any device level function call when possible.

File: include/drivers/fsw/driver.h (hunk 2 above)
Comment (wavious-jbasista) [reply include/drivers/fsw/driver.h:86]: Understood, you have identified the possible returns which would indicate which MSR is currently in use, so no issues here.

**Review by wavious-jbasista — APPROVED** [review]
Looks good to me, nice API change to show what MSR we are prepping for!

**Pull request merged**